src/utils/mtev_logic.c

```diff
@@ -320,10 +315,11 @@ static inline void *
 mtev_logic_ast_node_predicate(mtev_logic_ast_t *ast, const char *left,
                               mtev_logic_pred_op_t op, mtev_logic_var_t *right) {
   if(op == POP_RE || op == POP_NRE) {
+    char buff[4096];
```

Comment:
Should this be like, max metric name length constant?

Reply:
well, it doesn't have anything to do with a metric name, this is mtev and knows nothing about metrics.  It's just a sufficiently large buffer to use to stringify something... At some point we want to refuse to stringify b/c it would be too costly.